src/sys/wait.rs

```diff
@@ -215,24 +232,53 @@ fn decode(pid : Pid, status: i32) -> WaitStatus {
     }
 }
 
-pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Result<WaitStatus> {
+/// Waits for and returns events that are received from the given supplied process or process group
+/// ID, and associated options.
+///
+/// # Usage Notes
+///
+/// - If the value of PID is greater than `0`, it will wait on the child with that PID.
+/// - If the value of the PID is less than `-1`, it will wait on any child that
+///   belongs to the process group with the absolute value of the supplied PID.
+/// - If the value of the PID is `0`, it will wait on any child process that has the same
+/// group ID as the current process.
+/// - If the value of the PID is `-1`, it will wait on any child process of the current process.
+/// - If the value of the PID is `None`, the value of PID is set to `-1`.
```

Comment:
Combine this line with the above. But then, what's the point of making `Pid` and Option type rather than leaving it numeric if there are two `None` cases? In reality, this should take an enum of these different groups, which are then converted to a numeric `pid_t` internally. Like:

```
pub enum PidGroup {
    ChildPid(Pid),
    ProcessGroupChild(Pid),
    GroupChild,
    CurrentProcessChild,
}
```

Then the user uses one of those to specify the option. What do you think about that?